walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -93,6 +93,44 @@ private boolean nestedEventsCheck(TimeRange prevTimeRange, TimeRange currTimeRan
     return false;
   }
 
+  /**
+   * This method removes optional Events from iterator, timeRanges, if it contains mandatory attendees
+   * @param timeRanges iterator, contains all events 
+   * @param timeRangeEventMap HashMap object, utilize to find Event object based on given TimeRange
+   * @param optionalEvents Empty ArrayList, will contain all optional Events
+   * @param removeOpRanges Empty ArrayList, will contain all optional TimeRange objects
+   * @param request MeetingRequest object, utilized to check if Event is optional
+   * @param onlyOptionalEvents boolean, set to 'true' if timeRanges only contains optional events
+   * @return onlyOptionalEvents
+   */
+  private boolean processOptionalEvents(ArrayList<TimeRange> timeRanges,
+                                    HashMap<TimeRange, Event> timeRangeEventMap,
+                                    ArrayList<Event> optionalEvents,
+                                    ArrayList<TimeRange> removeOpRanges,
+                                    MeetingRequest request,
+                                    boolean onlyOptionalEvents){
```

Comment:
Is the `onlyOptionalEvents` parameter used?

Reply:
Nope, I forgot to delete it during development.